conda_forge_tick/migrators.xsh

```diff
@@ -0,0 +1,232 @@
+import copy
+import datetime
+import os
+import time
+import traceback
+import urllib.error
+
+import github3
+import networkx as nx
+from doctr.travis import run as doctr_run
+from pkg_resources import parse_version
+from rever.tools import (eval_version, indir, hash_url, replace_in_file)
+
+from conda_forge_tick.git_utils import (feedstock_url, feedstock_repo, fork_url,
+                                        get_repo)
+from conda_forge_tick.utils import parsed_meta_yaml
+
+
+class Migrator:
+    def filter(self, attrs):
+        """ If true don't act upon node
+
+        Parameters
+        ----------
+        attrs : dict
+            The node attributes
+
+        Returns
+        -------
+        bool :
+            True if node is to be skipped
+        """
+        # never run on archived feedstocks
+        return bool(attrs.get('archived', False))
+
+    def migrate(self, recipe_dir, attrs):
+        """Perform the migration, updating the ``meta.yaml``
+
+        Parameters
+        ----------
+        recipe_dir : str
+            The directory of the recipe
+        attrs : dict
+            The node attributes
+
+        Returns
+        -------
+
+        """
+        pass
+
+    def pr_body(self):
+        """Create a PR message body
+
+        Returns
+        -------
+        body: str
+            The body of the PR message
+        """
+        body = (
+            'This PR was created by the [cf-regro-autotick-bot](https://github.com/regro/cf-scripts).\n\n'
+            'The **cf-regro-autotick-bot** is a service to automatically '
+            'track the dependency graph, migrate packages, and '
+            'propose package version updates for conda-forge. '
+            'It is very '
+            'likely that the current package version for this feedstock is '
+            'out of date or needed migration.\n\n'
+            '{}'
+            "If you would like a local version of this bot, you might consider using "
+            "[rever](https://regro.github.io/rever-docs/). "
+            "Rever is a tool for automating software releases and forms the "
+            "backbone of the bot's conda-forge PRing capability. Rever is both "
+            "conda (`conda install -c conda-forge rever`) and pip "
+            "(`pip install re-ver`) installable.\n\n"
+            'Finally, feel free to drop us a line if there are any '
+            '[issues](https://github.com/regro/cf-scripts/issues)! ')
+        return body
+
+    def commit_message(self):
+        """Create a commit message"""
+        pass
```

Comment:
this should probably return a sensible default, such as `'migration: ' + self.__class__.__name__ `